packages/salesforcedx-vscode-soql/src/editor/soqlEditorProvider.ts

```diff
@@ -17,6 +17,14 @@ import {
 import { HtmlUtils } from './htmlUtils';
 import { SOQLEditorInstance } from './soqlEditorInstance';
 
+const sfdxCoreExtension = vscode.extensions.getExtension(
```

Comment:
Please don't use this pattern, we are working on removing this type of usage of the core extension since it's causing performance problems.

Reply:
@jonnyhork  for channelService, you can take a look at https://github.com/forcedotcom/salesforcedx-vscode/blob/f4515a439f737546e740ce8db749ffb6bfca5eec/packages/salesforcedx-vscode-lwc/src/commands/forceLightningLwcStart.ts#L14 and use the utils package

Reply:
@lcampos  Is there another way to get the `workspaceContext` or check for the default user name you would recommend?

If not, then the plan was for us to have to refactor our extension anyway once the work to expose this stuff in a utility is done on your end. So the core exports would be addressed then (we already use this pattern to get the connection).

We are hoping to have a releasable `vsix` by Nov 12th. Will the utilities be available before then?

Reply:
@xyc Looks like channelService is no longer available in the `out` dir when I add it as a dependency, did something change?
`'@salesforce/salesforcedx-utils-vscode/out/src/commands';`, commands does not exist.
I am honestly not sure, can I just use
`'@salesforce/salesforcedx-utils-vscode/src/commands';`?

Reply:
@jonnyhork actually let's wait a little bit for that, I think @lcampos still has some changes to be made on channelService

Reply:
Sounds good to me, I already gave up on it =)

Reply:
We will still need to know how to move forward from here. I realize we are trying to avoid using core as a dependency but my preference would be to move forward with whats accessable in core exports now so we can get the extension out the door. Then refactor this once the utility work is done.

We can also just stick this in waiting until the utils are available as long as its before Nov 12th.

Reply:
@jonnyhork you should be able to use channel service, notification service and telemetry from the utils module. Your branch is outdated so you should rebase first and so you can use those changes.

Specific to `workspaceContext`, that's not something we were thinking on immediately moving but should be simple to do now that we've done the heavy lifting needed in the utils module. We welcome you guys opening a PR for moving that to the utils module.

Reply:
Just getting caught up on this, looks like there is a work item on our board to submit a PR to pull out the worksapceContext to the utils, I think Jon Gellin will be doing this work. I may just put this work in waiting until that work is done. Still discussing with the team.